lib/error.js

```diff
@@ -2,9 +2,47 @@ var makeError = require('make-error');
 
 function ShareDBError(code, message) {
   ShareDBError.super.call(this, message);
-  this.code = code;
+  this.code = code || ShareDBError.code.UNKNOWN_ERROR;
 }
 
 makeError(ShareDBError);
 
+ShareDBError.code = {
+  BADLY_FORMED_MESSAGE: 'ERR_BADLY_FORMED_MESSAGE',
+  CANNOT_PROJECT_TYPE: 'ERR_CANNOT_PROJECT_TYPE',
+  CREATE_DATA_MUST_BE_AN_OBJECT: 'ERR_CREATE_DATA_MUST_BE_AN_OBJECT',
+  DATABASE_ADAPTER_NOT_FOUND: 'ERR_DATABASE_ADAPTER_NOT_FOUND',
+  DEL_MUST_BE_TRUE: 'ERR_DEL_MUST_BE_TRUE',
+  DOCUMENT_ALREADY_CREATED: 'ERR_DOCUMENT_ALREADY_CREATED',
+  DOCUMENT_CREATED_REMOTELY: 'ERR_DOCUMENT_CREATED_REMOTELY',
+  DOCUMENT_DOES_NOT_EXIST: 'ERR_DOCUMENT_DOES_NOT_EXIST',
+  DOCUMENT_WAS_DELETED: 'ERR_DOCUMENT_WAS_DELETED',
+  INVALID_CLIENT_ID: 'ERR_INVALID_CLIENT_ID',
+  INVALID_DEFAULT_TYPE: 'ERR_INVALID_DEFAULT_TYPE',
+  INVALID_METADATA: 'ERR_INVALID_METADATA',
+  INVALID_SRC: 'ERR_INVALID_SRC',
```

Comment:
Let's group `INVALID_SRC`, `INVALID_SEQ`, and `SRC_AND_SEQ...` as `INVALID_OP_FORMAT`, and add more detail as the message.